core/src/jmh/java/com/uber/m3/tally/AbstractReporterBenchmark.java

```diff
@@ -107,40 +137,24 @@ public void reportHistogramValueSamplesBenchmark(IncrementalValueState value) {
         );
     }
 
-    @Setup
+    @Setup(Level.Iteration)
     public void setup() {
-        this.reporter = initReporter();
-
-        reporter.reportCounter(COUNTER_NAME, DEFAULT_TAGS, 1);
-        reporter.reportGauge(GAUGE_NAME, DEFAULT_TAGS, 1);
-        reporter.reportTimer(TIMER_NAME, DEFAULT_TAGS, Duration.ofSeconds(1));
-        reporter.reportHistogramValueSamples(
-                HISTOGRAM_VALUE_NAME,
-                DEFAULT_TAGS,
-                VALUE_EXPONENTIAL_BUCKETS,
-                0.1,
-                0.1,
-                1
-        );
-        reporter.reportHistogramDurationSamples(
-                HISTOGRAM_DURATION_NAME,
-                DEFAULT_TAGS,
-                DURATION_EXPONENTIAL_BUCKETS,
-                Duration.ofSeconds(1),
-                Duration.ofSeconds(1),
-                1
-        );
+        this.reporter = bootReporter();
+        logger.info("Booted reporter");
     }
 
-    @TearDown
+    @TearDown(Level.Iteration)
     public void teardown() {
-        reporter.close();
+        shutdownReporter(reporter);
+        reporter = null;
+        logger.info("Shutdown reporter");
     }
 
-    public abstract StatsReporter initReporter();
+    public abstract Reporter bootReporter();
+    public abstract void shutdownReporter(Reporter reporter);
 
     @State(Scope.Thread)
-    public static class IncrementalValueState {
+    public static class Counter {
```

Comment:
rename it to Counter makes sense but it melts with the name of the Counter reporter. I don't have a strong preferences, your call if you want to change it.

Reply:
Not sure i understand which one you're referring to